resolver/resolver.cc

```diff
@@ -1447,28 +1447,26 @@ class ResolveSignaturesWalk {
     void injectOptionalArgs(core::MutableContext ctx, ast::MethodDef *mdef) {
         ast::InsSeq::STATS_store lets;
 
-        if (mdef->symbol.data(ctx)->isAbstract()) {
-            // If we checked that abstract methods don't have defined bodies earlier (currently done in infer)
-            // we could also use this technique to check default arguments of abstract methods.
-            return;
-        }
-
         int i = -1;
         for (auto &argSym : mdef->symbol.data(ctx)->arguments()) {
             i++;
             auto &argExp = mdef->args[i];
             auto argType = argSym.type;
 
             if (auto *optArgExp = ast::cast_tree<ast::OptionalArg>(argExp.get())) {
-                // Using optArgExp's loc will make errors point to the arg list, even though the T.let is in the
-                // body.
-                auto let = make_unique<ast::Cast>(optArgExp->loc, argType, optArgExp->default_->deepCopy(),
-                                                  core::Names::let());
-                lets.emplace_back(std::move(let));
+                if (!argType) {
+                    optArgExp->default_ = nullptr;
```

Comment:
Why throw the default value away when there's no argument type? Couldn't you make a normal assignment instead of a `T.let`?

Reply:
Interesting, for what reason? To find `def foo(a = 1 + "2")`?

Reply:
Yes, if the file was `# typed: true`, but the method was missing a sig, it would be nice to catch this case.